src/client.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for interacting with the GoCert application."""
+
+import requests
+
+
+class GoCertClientError(Exception):
+    """Base class for exceptions raised by the GoCert client."""
+
+
+class GoCert:
+    """Class to interact with GoCert."""
+
+    def __init__(self, host: str, ca_path: str) -> None:
+        self.host = host
+        self.ca_path = ca_path
+
+    def is_api_available(self) -> bool:
+        """Return if the GoCert server is reachable."""
+        try:
+            req = requests.get(
+                f"https://{self.host}/status",
```

Comment:
Here we assume the scheme, I would encourage having this be provided by the user of the class:

```python
client = Gocert(url="https://gocert.com")
```

Also, in the init we can give an example of what we are expecting here in terms of format

Reply:
done, and added docstring to init